cmd/metrics-v3-cache.go

```diff
@@ -61,8 +62,15 @@ func newNodesUpDownCache() *cachevalue.Cache[nodesOnline] {
 		loadNodesUpDown)
 }
 
+type diskIOStats struct {
+	stats    map[string]madmin.DiskIOStats
+	duration time.Duration
+}
+
+// storageMetrics - cached storage metrics.
 type storageMetrics struct {
 	storageInfo                              madmin.StorageInfo
+	ioStats                                  *diskIOStats
```

Comment:
The current implementation is also fine, Just a suggestion here in case if you think it is better.

Can we fix the types to be like the following

```go
type perDriveIOStats struct {
     driveReadsPerSec  float64
     driveReadsKBPerSec float64
     driveReadsAwait float64
     driveWritesPerSec float64
     driveWritesKBPerSec float64
     driveWritesAwait float64
     drivePercUtil float64
}
```

```go
type storageMetrics struct {
     ...
     ioStats      map[string]perDriveIOStats
     ...
}
```
and keep all the calculations here in `loadDriveMetrics()` and just set the calculated values in `setDriveIOStatMetrics()`

Reply:
There aren't any benefits in the above approach, just that the calculations will be on one side and setMetrics() will just set the metrics and will not do any calculations. This won't spill the calculation logic to more than one place.

Reply:
@Praveenrajmani I liked this suggestion. Have implemented, just with slightly different names for the struct and its fields. PTAL.